modules/selenium/src/main/java/org/testcontainers/containers/BrowserWebDriverContainer.java

```diff
@@ -193,39 +202,41 @@ public RemoteWebDriver getWebDriver() {
 
     @Override
     protected void failed(Throwable e, Description description) {
-        switch (recordingMode) {
-            case RECORD_FAILING:
-            case RECORD_ALL:
-                stopAndRetainRecordingForDescriptionAndSuccessState(description, false);
-                break;
-        }
-        currentVncRecordings.clear();
+        stopAndRetainRecordingForDescriptionAndSuccessState(description, false);
     }
 
     @Override
     protected void succeeded(Description description) {
-        switch (recordingMode) {
-            case RECORD_ALL:
-                stopAndRetainRecordingForDescriptionAndSuccessState(description, true);
-                break;
-        }
-        currentVncRecordings.clear();
+        stopAndRetainRecordingForDescriptionAndSuccessState(description, true);
     }
 
     @Override
-    protected void finished(Description description) {
+    public void stop() {
         if (driver != null) {
-            driver.quit();
+            try {
+                driver.quit();
+            } catch (Exception e) {
+                LOGGER.debug("Failed to quit the driver", e);
+            }
         }
-        this.stop();
+
+        if (vncRecordingContainer != null) {
+            try {
+                vncRecordingContainer.stop();
+            } catch (Exception e) {
+                LOGGER.debug("Failed to stop vncRecordingContainer", e);
+            }
+        }
+
+        super.stop();
     }
 
     private void stopAndRetainRecordingForDescriptionAndSuccessState(Description description, boolean succeeded) {
-        File recordingFile = recordingFileFactory.recordingFileForTest(vncRecordingDirectory, description, succeeded);
-        LOGGER.info("Screen recordings for test {} will be stored at: {}", description.getDisplayName(), recordingFile);
+        if (recordingMode == VncRecordingMode.RECORD_ALL || (!succeeded && recordingMode == VncRecordingMode.RECORD_FAILING)) {
```

Comment:
I would prefer a variable for `(!succeeded && recordingMode == VncRecordingMode.RECORD_FAILING)`, since I initially needed some time to parse this boolean expression 😉 
Maybe `recordFailingTest`?

Reply:
Good idea 👍